zephyr/esp32/CMakeLists.txt

```diff
@@ -4,7 +4,9 @@ if(CONFIG_SOC_SERIES_ESP32)
 
   zephyr_compile_options(-fstrict-volatile-bitfields)
   zephyr_compile_definitions_ifndef(CONFIG_MCUBOOT CONFIG_APP_BUILD_USE_FLASH_SECTIONS)
-  zephyr_compile_definitions_ifndef(CONFIG_MCUBOOT BOOTLOADER_BUILD=y) # See Zephyr-ESP / ZEP-820
+
+  zephyr_compile_definitions_ifdef(CONFIG_MCUBOOT BOOTLOADER_BUILD)
+  zephyr_compile_definitions_ifdef(CONFIG_ESP_SIMPLE_BOOT BOOTLOADER_BUILD)
```

Comment:
Having `CONFIG_ESP_SIMPLE_BOOT` here sounds wrong.

Reply:
> Having `CONFIG_ESP_SIMPLE_BOOT` here sounds wrong. Why only for ESP32?

That is because the `BOOTLOADER_BUILD` is delimiting the DPORT access and that caused the startup issue, so we need to enable this flag here. Other SoCs don't have this kind of issue and no flag needs to be defined. However, we will remove that flag in the near future and this line will go away.